Enhance inventory sorting with `SortableJS`

In3luki opened the pull request:
[sortable-inventory.webm](https://user-images.githubusercontent.com/41452412/229311409-0ae583dc-e78e-4f05-8527-3c0d6271a8ad.webm)

stwlam commented on src/module/actor/sheet/base.ts:
```diff
@@ -589,6 +598,80 @@ abstract class ActorSheetPF2e<TActor extends ActorPF2e> extends ActorSheet<TActo
         });
 
+        // Inventory sorting
+        const inventoryList = htmlQueryAll(html, "section.inventory-list, ol[data-container-type=actorInventory]").find(
+            (e) => e !== null
+        );
```
Shouldn't this just be an `htmlQuery`? `htmlQueryAll` can't return an array with `null` elements.

In3luki replied:
You're right. I was going from the assumption that two selectors needed to be resolved with ``htmlQueryAll` but `htmlQuery` works just fine.

stwlam commented:
Could dropping an item directly onto a container row (rather than its contents) easily be handled to place it in the target container?

stwlam commented on src/module/item/physical/document.ts:
```diff
@@ -318,6 +319,106 @@ abstract class PhysicalItemPF2e<TParent extends ActorPF2e | null = ActorPF2e | n
+        toContainer?: ContainerPF2e<ActorPF2e> | null;
+        render?: boolean;
+    }): Promise<void> {
+        if (!this.actor) {
+            throw ErrorPF2e(`Tried to move an unonwned item!`);
+        }
+        if (!toContainer && !relativeTo) {
+            throw ErrorPF2e(`You must provide either the relativeTo or the toContainer parameter!`);
+        }
+        if (relativeTo) {
+            if (!relativeTo.isOfType("physical")) {
+                throw ErrorPF2e(`The relativeTo item is not a physical item!`);
+            } else if (!relativeTo.actor) {
+                throw ErrorPF2e(`The relativeTo item is not an owned item!`);
+            }
+
+            // If theses two items are stackable, combine them
+            if (await this.stackWith(relativeTo)) return;
+        }
+        if (toContainer) {
+            if (!toContainer.isOfType("backpack")) {
+                throw ErrorPF2e(`The toContainer item is not a container item!`);
+            } else if (!toContainer.actor) {
+                throw ErrorPF2e(`The toContainer item is not an owned item!`);
+            }
+        }
```
I suspect most of this could be handled with type checking

In3luki replied:
This is intended for modules or macros that might pass in the wrong item types.

In3luki commented:
> Could dropping an item directly onto a container row (rather than its contents) easily be handled to place it in the target container?

Fixed. It was working before but must have broken when I added item stacking.
- Dropping on a container puts the item into the container.
- Dropping in container contents puts the item relative to the contained item.

In3luki commented:
I *hope* everything is working now:

[sortable-inventory.webm](https://user-images.githubusercontent.com/41452412/230380733-cb895a8b-2a70-4769-8507-4c7cf16ee1a4.webm)
